src/conventions/h-group/take-action.js

```diff
@@ -230,7 +269,57 @@ export function take_action(game) {
 	let best_play_clue, clue_value;
 	if (state.clue_tokens > 0) {
 		const all_play_clues = play_clues.flat();
-		({ clue: best_play_clue, clue_value } = select_play_clue(all_play_clues));
+		let consider_clues = all_play_clues;
+		const chop = game.me.chop(state.hands[state.ourPlayerIndex]);
+		let saved_clue;
+		let saved_clue_value = -99;
+
+		// Consider saving clues to finesse positions for players who likely have
+		// better cards on chop.
+		if (!state.inEndgame() && !state.early_game && state.clue_tokens < 4 && chop !== undefined) {
+			const our_chop_value = cardValue(state, game.me, chop, chop.order);
+			const better_givers = [];
+			for (let i = 0; i < state.numPlayers; ++i) {
+				if (i == state.ourPlayerIndex)
+					continue;
+				const player = game.players[i];
+				const hand = state.hands[i];
+				if (player.thinksLoaded(state, i, {assume: false}))
+					continue;
+				const otherChop = player.chop(hand);
+				// Saves clue for locked players or players who likely have a better chop.
+				if (otherChop === undefined && player.thinksLocked(state, i))
+					better_givers.push(i);
+				else if (otherChop !== undefined && cardValue(state, player, otherChop, otherChop.order) >= our_chop_value)
+					better_givers.push(i);
+			}
+			if (better_givers.length > 0) {
+				let saved_for = [];
+				consider_clues = consider_clues.filter(clue => {
+					const list = state.hands[clue.target].clueTouched(clue, state.variant).map(c => c.order);
+					const { finesse } = determine_focus(state.hands[clue.target], common, list, {beforeClue: true});
+					if (!finesse)
+						return true;
+
+					const save_for = find_clue_givers(game, clue, state.ourPlayerIndex).filter(playerIndex => better_givers.includes(playerIndex));
+					if (save_for.length == 0)
+						return true;
+					const value = find_clue_value(clue.result);
+					if (saved_clue === undefined || value > saved_clue_value) {
+						saved_for = save_for;
+						saved_clue = clue;
+						saved_clue_value = value;
+					}
+					return false;
+				});
+				if (saved_clue !== undefined)
+					logger.info(`saved clue ${logClue(saved_clue)} for ${saved_for.map(playerIndex => state.playerNames[playerIndex]).join(', ')}`);
+			}
+		}
+
+		({ clue: best_play_clue, clue_value } = select_play_clue(consider_clues));
+		if (saved_clue !== undefined && saved_clue_value > clue_value && state.clue_tokens < 2)
```

Comment:
If we require the clue count to be < 2 in order to consider leaving the clue for someone else, why do we do the above calculations when the clue count is < 4? Is it that when there are 2-3 clues, you still want to avoid giving the clues that could be better given by someone else, but prefer giving an unrelated play clue over discarding? Whereas if the clue count is < 2 (and there is a clue someone would be good to give), we don't want to give any play clue?

Seems reasonable enough I guess, if a little arbitrary. Would have to see how it plays out.

Reply:
> If we require the clue count to be < 2 in order to consider leaving the clue for someone else

If the clue count is < 4 we have already filtered out the clue that should be left for someone else. If the clue count is less than 2, i.e. 1, then if we use the last clue on something that isn't the best clue that we saved for someone else then they won't be able to give the best clue that we didn't give because we filtered it out.

> , why do we do the above calculations when the clue count is < 4? Is it that when there are 2-3 clues, you still want to avoid giving the clues that could be better given by someone else, but prefer giving an unrelated play clue over discarding?

That's exactly right. It's still allowing giving an unrelated clue if there are others to give. TBD to ensure that the unrelated clue doesn't interfere with that clue.

> Whereas if the clue count is < 2 (and there is a clue someone would be good to give), we don't want to give any play clue?

Yes, so that there's still a clue token to give the best clue.

> Seems reasonable enough I guess, if a little arbitrary. Would have to see how it plays out.

The < 4 is a bit arbitrary, but I figured if you have lots of clues it's probably better to just give the clue.